corehq/form_processor/utils/sql.py

```diff
@@ -212,3 +212,56 @@ def prepare(self, conn):
         for field in self.fields:
             if hasattr(field, 'prepare'):
                 field.prepare(conn)
+
+
+def get_case_transactions_by_case_id(case_id, updated_xforms=None):
+    """
+    This fetches all the transactions required to rebuild the case along
+    with all the forms for those transactions.
+
+    For any forms that have been updated it replaces the old form
+    with the new one.
+
+    :param case_id: ID of case to rebuild
+    :param updated_xforms: list of forms that have been changed.
+    :return: list of ``CaseTransaction`` objects with their associated forms attached.
+    """
+
+    transactions = CaseAccessorSQL.get_transactions_for_case_rebuild(case_id)
+    return fetch_case_transaction_forms(transactions, updated_xforms)
+
+
+def fetch_case_transaction_forms(transactions, updated_xforms=None):
+    """
+    Fetches the forms for a list of transactions
+
+    :param transactions: list of ``CaseTransaction`` objects:
+    :param updated_xforms: list of forms that have been changed.
+    :return: list of ``CaseTransaction`` objects with their associated forms attached.
+    """
+
+    form_ids = {tx.form_id for tx in transactions}
+    updated_xforms_map = {
+        xform.form_id: xform for xform in updated_xforms if not xform.is_deprecated
+    } if updated_xforms else {}
+
+    form_ids_to_fetch = list(form_ids - set(updated_xforms_map.keys()))
```

Comment:
`.keys()` is not necessary and will make the Python3 conversion utils complain and want to wrap it with `list()`.